tests/common/system_utils/docker.py

```diff
@@ -0,0 +1,157 @@
+"""
+    docker contains utilities for interacting with Docker on the DUT.
+"""
+
+import collections
+import logging
+import os
+import time
+import yaml
+
+from common.broadcom_data import is_broadcom_device
+from common.mellanox_data import is_mellanox_device
+
+_LOGGER = logging.getLogger(__name__)
+
+_LOGIN_CMD = "docker login -u {} -p {} {}"
+_PULL_CMD = "docker pull {}/{}:{}"
+_RM_CMD = "docker rm {}"
+_TAG_CMD = "docker tag {}:{} {}"
+
+_BASE_DIR = os.path.dirname(os.path.realpath(__file__))
+SONIC_DOCKER_REGISTRY = os.path.join(_BASE_DIR, "../../../ansible/vars/docker_registry.yml")
+
+_DockerRegistryInfo = collections.namedtuple('DockerRegistryInfo', 'host username password')
+class DockerRegistryInfo(_DockerRegistryInfo):
+    """
+        DockerRegistryInfo holds all the data needed to access a remote Docker registry.
+
+        Attributes:
+            host (str): The remote host where the Docker registry is located.
+            username (str): The username used to access the registry.
+            password (str): The password used to access the registry.
+    """
+    pass
+
+def parse_registry_file(registry_file):
+    """
+        parse_registry_file parses the provided file to produce a DockerRegistryInfo.
+
+        See `SONIC_DOCKER_REGISTRY` for the expected format of this file.
+
+        Args:
+            registry_file (str): The name of the file holding the registry information.
+
+        Raises:
+            IOError: If the file cannot be opened for any reason.
+            ValueError: If the provided file is missing any required fields.
+
+        Returns:
+            DockerRegistryInfo: The registry info from the registry file.
+    """
+
+    try:
+        with open(registry_file) as contents:
+            registry_vars = yaml.safe_load(contents)
+    except IOError as err:
+        _LOGGER.error("Failed to parse registry file \"%s\" (%s)", registry_file, err)
+        raise
+
+    host = registry_vars.get("docker_registry_host")
```

Comment:
there are docker modules in Ansible. It is worth investigating using them. They will hide the password, I assume.

https://docs.ansible.com/ansible/latest/modules/docker_login_module.html#docker-login-module

Reply:
Oh cool, thanks for sharing! I'll take a look. :)